### PR TITLE
Remove legacy PHP versions and add type declarations

### PR DESCRIPTION
**Pull request by johnstevenson**
This is a full-monty update to modern PHP and will require a major version bump to 3.

I tried to keep things compatible with version 2, but although it added some value it just felt like I was kicking the can down the road.

For the majority of dependents (~94% from Packagist) the strict typing will have absolutely no effect as they do no more than call `XdebugHandler::check`. The remaining 4 projects extend the main class so that they can restart for other reasons. Two of these already return-typehint their overridden methods so this version will work out of the box, one can easily be updated and the other still supports PHP 5.3 so it can't be.

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -9,19 +9,9 @@
...
-
     <testsuites>
         <testsuite name="XdebugHandler Test Suite">
             <directory>tests</directory>
         </testsuite>
     </testsuites>
-
-    <filter>
-        <whitelist>
-            <directory>src</directory>
```
Comment (stof) [inline phpunit.xml.dist:24]: The whitelist config should not be removed, otherwise generating code coverage becomes impossible.

Comment (johnstevenson) [reply phpunit.xml.dist:24]: Yeah, it's a bit brutal but the `<filter>` element has been replaced by `<coverage>` in PHPUnit 9.

You can still generate code coverage though with the appropriate command line option (`--whitelist <dir>` or `--coverage-filter <dir>`).

Comment (stof) [reply phpunit.xml.dist:24]: Well, PHPUnit 9 still supports the old format

**Review by Seldaek — APPROVED** [review]
Looks good to me, although doing this means 2.x will still have to be maintained if any blocking bug comes up due to the Composer 2.2 LTS dependency. But I don't imagine this is likely to cause lots of overhead on this project.

**Comment by johnstevenson** [comment]
@Seldaek Thanks.

> means 2.x will still have to be maintained if any blocking bug comes up due to the Composer 2.2 LTS dependency.

Absolutley, that's the plan and I agree that I don't think this will be too much of a burden.

**Pull request merged**